apps/passport-client/src/zapp/ZappServer.ts

```diff
@@ -270,6 +270,43 @@ class ZupassPODRPC extends BaseZappServer implements ParcnetPODRPC {
       this.getAdvice().showClient();
     });
   }
+
+  public async signPrefixed(entries: PODEntries): Promise<PODData> {
+    const origin = this.getContext().getState().zappOrigin;
+    if (
+      appConfig.zappRestrictOrigins &&
+      (!origin || !appConfig.zappAllowedSignerOrigins.includes(origin))
+    ) {
+      throw new Error("Origin not allowed to sign PODs");
+    }
+    if (!this.getPermissions().SIGN_POD) {
+      throw new MissingPermissionError("SIGN_POD", "pod.sign");
+    }
+
+    for (const name of Object.keys(entries)) {
+      if (!name.startsWith("_UNSAFE_")) {
+        throw new Error(
+          "PODs signed with signPrefixed must have a prefix of _UNSAFE_"
+        );
+      }
+    }
+
+    entries.UNSAFE_META_ORIGIN = {
```

Comment:
Why all caps?
It's a bit odd that this starts with UNSAFE_ not _UNSAFE_ given the above, but I guess that's intentional that you want it to be non-overlapping with user-specified entries (as opposed to overriding one?).